scrape/scrape.go

```diff
@@ -1461,13 +1461,28 @@ func (sl *scrapeLoop) append(app storage.Appender, b []byte, contentType string,
 	}
 
 	var (
-		defTime        = timestamp.FromTime(ts)
-		appErrs        = appendErrors{}
-		sampleLimitErr error
-		e              exemplar.Exemplar // escapes to heap so hoisted out of loop
-		meta           metadata.Metadata
+		defTime         = timestamp.FromTime(ts)
+		appErrs         = appendErrors{}
+		sampleLimitErr  error
+		e               exemplar.Exemplar // escapes to heap so hoisted out of loop
+		meta            metadata.Metadata
+		metadataChanged bool
 	)
 
+	checkMetadataUpdates := func(lset labels.Labels, isNewSeries bool) {
+		if sl.appendMetadataToWAL {
+			sl.cache.metaMtx.Lock()
+			metaEntry, metaOk := sl.cache.metadata[yoloString([]byte(lset.Get(labels.MetricName)))]
+			if metaOk && (isNewSeries || metaEntry.lastIterChange == sl.cache.iter) {
```

Comment:
can we leave a comment? `<function> updates the metadata fields and metadataChanged value if we have metadata in the scrape cache AND the labelset is for a new series or the metadata for this series has changed`
the function name should also maybe be `updateMetadata` and could return a bool of whether or not it changed the metadata

Reply:
You're right, the comment is a good idea let me add it.

Do we have a use case for returning that bool here in the scrape loop, or would we just ignore it? Or would it be for example useful for tests? It's not 100% clear to me right now.